Source/DafnyCore/AST/DafnyAst.cs

```diff
@@ -31,6 +31,14 @@ public interface INode {
     /// losing source location information, so those transformed nodes should not be returned by this property.
     /// </summary>
     IEnumerable<INode> Children { get; }
+
+    /// <summary>
+    /// These tokens are "owned" by the node, roughly meaning
+    /// they are not "owned" by any other node normally (typically children)
+    /// Owned tokens will help the formatter determine formatting cues
```

Comment:
I wouldn't mention the formatter since these tokens can be used for other things as well, such as simply reprinting the program, something related to documentation on hover perhaps, generated a documentation page.